pkg/kn/commands/revision/describe.go

```
@@ -46,19 +62,225 @@ func NewRevisionDescribeCommand(p *commands.KnParams) *cobra.Command {
+	target := clientserving.ConcurrencyTarget(&revision.ObjectMeta)
+	limit := revision.Spec.ContainerConcurrency
+	if target != nil || limit != nil {
+		section := dw.WriteAttribute("Concurrency", "")
+		if limit != nil {
+			section.WriteAttribute("Limit", strconv.FormatInt(int64(*limit), 10))
+		}
+		if target != nil {
+			section.WriteAttribute("Target", strconv.Itoa(*target))
+		}
+	}
+}
+
+// Write the image attribute (with
+func WriteImage(dw printers.PrefixWriter, revision *v1alpha1.Revision) {
+	c, err := clientserving.ContainerOfRevisionSpec(&revision.Spec)
+	if err != nil {
+		dw.WriteAttribute("Image", "Unknown")
+		return
+	}
+	image := c.Image
+	// Check if the user image is likely a more user-friendly description
+	pinnedDesc := "at"
+	userImage := clientserving.UserImage(&revision.ObjectMeta)
+	imageDigest := revision.Status.ImageDigest
+	if userImage != "" && imageDigest != "" {
+		var parts []string
+		if strings.Contains(image, "@") {
+			parts = strings.Split(image, "@")
+		} else {
+			parts = strings.Split(image, ":")
+		}
+		// Check if the user image refers to the same thing.
+		if strings.HasPrefix(userImage, parts[0]) {
+			pinnedDesc = "pinned to"
+			image = userImage
+		}
+	}
+	if imageDigest != "" {
+		image = fmt.Sprintf("%s (%s %s)", image, pinnedDesc, shortenDigest(imageDigest))
+	}
+	dw.WriteAttribute("Image", image)
+}
+
+func WritePort(dw printers.PrefixWriter, revision *v1alpha1.Revision) {
+	port := clientserving.Port(&revision.Spec)
+	if port != nil {
+		dw.WriteAttribute("Port", strconv.FormatInt(int64(*port), 10))
+	}
+}
+
+func WriteEnv(dw printers.PrefixWriter, revision *v1alpha1.Revision, printDetails bool) {
+	env := stringifyEnv(revision)
+	if env != nil {
+		commands.WriteSliceDesc(dw, env, "Env", printDetails)
+	}
+}
+
+func WriteScale(dw printers.PrefixWriter, revision *v1alpha1.Revision) {
+	// Scale spec if given
+	scale, _ := clientserving.ScalingInfo(&revision.ObjectMeta)
```

Don't want to nag but ignoring errors here... could easily be handled IMO.